src/boss/mar/17FEC.c

```diff
@@ -1,7 +1,179 @@
+        D_80097928 = 1;
+        g_Statues[RIGHT_STATUE] = false; // right statue closed
+        self->animSet = ANIMSET_OVL(1);
+        self->animCurFrame = 23;
+        D_80097910 = 0;
+        self->zPriority = 0x40;
+
+        // Create clock hands
+        newEntity = &self[5];
+        for (i = 0; i < 2; i++) {
+            CreateEntityFromCurrentEntity(E_CLOCK_HANDS, newEntity);
+            newEntity->params = i;
+            newEntity++;
+        }
+
+        // Clock set
+        self[minuteHand].ext.clockRoom.hand = status->timerMinutes * 0x3C;
+        self[hourHand].ext.clockRoom.hand =
+            (status->timerHours * 0x12C) + (status->timerMinutes * 5);
+
+        // Create Birdcage doors
+        newEntity = &self[birdcageDoor1];
+        for (i = 0; i < 2; i++) {
+            CreateEntityFromCurrentEntity(E_BIRDCAGE_DOOR, newEntity);
+            newEntity->params = i;
+            newEntity++;
+        }
+
+        if ((status->timerMinutes >= 10) && (status->timerMinutes < 30)) {
+            self[birdcageDoor1].ext.birdcage.state = true;
+        } else {
+            self[birdcageDoor1].ext.birdcage.state = false;
+        }
+
+        if ((status->timerMinutes >= 30) && (status->timerMinutes < 50)) {
+            self[birdcageDoor2].ext.birdcage.state = true;
+        } else {
+            self[birdcageDoor2].ext.birdcage.state = false;
+        }
+
+        // Shadow for the Bighorn sheep head on the center
+        CreateEntityFromCurrentEntity(E_DUMMY_1D, &self[shadow]);
+        self[shadow].animSet = ANIMSET_OVL(1);
+        self[shadow].animCurFrame = 23;
+        self[shadow].zPriority = 0x40;
+        self[shadow].palette = 0x804B;
+        self[shadow].drawFlags = FLAG_DRAW_UNK8;
+        self[shadow].blendMode = 0x10;
+        self[shadow].flags = FLAG_DESTROY_IF_OUT_OF_CAMERA |
+                             FLAG_POS_CAMERA_LOCKED | FLAG_KEEP_ALIVE_OFFCAMERA;
+        self[shadow].posY.i.hi += 4;
+
+        // Create path blocking statues
+        newEntity = &self[statues];
+        for (i = 0; i < 2; i++) {
+            CreateEntityFromCurrentEntity(E_STATUE, newEntity);
+            newEntity->params = i;
+            newEntity++;
+        }
+
+        // Create the gears that drive the statues
+        newEntity = &self[gears];
+        for (j = 0; j < 2; j++) {
+            CreateEntityFromCurrentEntity(E_STATUE_GEAR, newEntity);
+            newEntity->params = j;
+            newEntity++;
+        }
+
+        // Create the stones on the floor
+        newEntity = &self[stoneDoors];
+        for (j = 0; j < 2; j++) {
+            CreateEntityFromCurrentEntity(E_STONE_DOOR, newEntity);
+            newEntity->params = j;
+            newEntity++;
+        }
+        break;
+
+    case 1:
+        if (status->timerFrames == 0) {
+            g_api.PlaySfx(0x7A9); // clock tick
+        }
+
+        // Update clock hands
+        self[minuteHand].ext.clockRoom.hand = status->timerMinutes * 0x3C;
```

Comment:
```suggestion
self[minuteHand].ext.clockRoom.hand = status->timerMinutes * 60;
```

Reply:
I thought of this as a clock handle angle before, but Sozud recommended it to be changed to 60 in the original PR. Your choice.